components/scream/src/physics/cld_fraction/cldfraction_main_impl.hpp

```diff
@@ -0,0 +1,88 @@
+#ifndef CLDFRACTION_MAIN_IMPL_HPP
+#define CLDFRACTION_MAIN_IMPL_HPP
+
+#include "physics/cld_fraction/cldfraction_functions.hpp"
+#include "ekat/kokkos/ekat_subview_utils.hpp"
+
+namespace scream {
+namespace cldfrac {
+
+/*-----------------------------------------------------------------*/
+template <typename S, typename D>
+void Functions<S,D>
+::cldfraction_main(
+  const Int nj,
+  const Int nk,
+  const view_2d<const Spack>& qi,
+  const view_2d<const Spack>& alst,
+  const view_2d<Spack>& aist,
+  const view_2d<Spack>& ast)
+{
+  using ExeSpace = typename KT::ExeSpace;
+  const Int nk_pack = ekat::npack<Spack>(nk);
+  const auto policy = ekat::ExeSpaceUtils<ExeSpace>::get_default_team_policy(nj, nk_pack);
+  Kokkos::parallel_for(
+    "cld fraction main loop",
+    policy,
+    KOKKOS_LAMBDA(const MemberType& team) {
+
+    const Int i = team.league_rank();
+
+    const auto oqi   = ekat::subview(qi,   i);
+    const auto oalst = ekat::subview(alst, i);
+    const auto oaist = ekat::subview(aist, i);
+    const auto oast  = ekat::subview(ast,  i);
+
+    cldfraction_calc_icefrac(team,nk,oqi,oaist);
+
+    cldfraction_calc_totalfrac(team,nk,oalst,oaist,oast);
+  });
+  Kokkos::fence();
+} // cldfraction_main
+/*-----------------------------------------------------------------*/
+template <typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::cldfraction_calc_icefrac(
```

Comment:
Do we need the cldfraction_ bit in the front of this and the next name? Isn't it obvious that this is a cloud fraction function from the rest of its name and because it is in the cldfrac namespace?

Reply:
Maybe we should rename the enclosing class: `Functions`->`CldFractionFunctions`.

Reply:
I like that idea.

Reply:
I went ahead and did this for the most recent version